### PR TITLE
DOCSP-45311: Remove deprecated utf8 types

### PR DESCRIPTION
**Pull request by norareidy**
# Pull Request Info

[PR Reviewing Guidelines](https://github.com/mongodb/docs-java/blob/master/REVIEWING.md)

JIRA - https://jira.mongodb.org/browse/DOCSP-45311
Staging - https://deploy-preview-78--docs-cpp.netlify.app/upgrade/#version-4.0

## Self-Review Checklist

- [ ] Is this free of any warnings or errors in the RST?
- [x] Did you run a spell-check?
- [x] Did you run a grammar-check?
- [x] Are all the links working?
- [ ] Are the [facets and meta keywords](https://wiki.corp.mongodb.com/display/DE/Docs+Taxonomy) accurate?

### CODE REVIEW

File: source/upgrade.txt [hunk 1]
```diff
@@ -81,6 +81,17 @@ This driver version introduces the following breaking changes:
   - ``BSONCXX_POLY_USE_SYSTEM_MNMLSTC``.
   - ``BSONCXX_POLY_USE_BOOST``.
 
+- Removes the following deprecated ``utf8`` types that have ``string`` equivalents:
+
+  - ``k_utf8`` from the ``bsoncxx::v_noabi::type`` class. Instead, use ``k_string``.
+  - ``b_utf8`` from the ``bsoncxx::v_noabi::types`` class. Instead, use ``b_string``.
+  - ``get_utf8`` from the ``bsoncxx::v_noabi::document::element``, ``bsoncxx::v_noabi::array::element``,
+    and ``bsoncxx::v_noabi::types::bson_value::view`` classes. Instead, use ``get_string``.
```
Comment (rustagir) [inline source/upgrade.txt:89]: Q: is this a type or a method?

File: source/whats-new.txt [hunk 2]
```diff
@@ -102,6 +102,17 @@ What's New in 4.0
      library equivalents. For context, see the :ref:`Build System Breaking Changes
      <version-4.0-build-system-breaking>` admonition.
 
+   - Removes the following deprecated ``utf8`` types that have ``string`` equivalents:
```
Comment (rustagir) [inline source/whats-new.txt:105]: S: you could consider removing the full list from here and just linking to the upgrade guide for specific details and keep the list there?

**Review by rustagir — APPROVED** [review]
lgtm with some qs

File: source/upgrade.txt (hunk 1 above)
Comment (norareidy) [reply source/upgrade.txt:89]: That one's a method! fixed

File: source/upgrade.txt [hunk 3]
```diff
@@ -151,6 +151,17 @@ The v4.0 driver introduces the following breaking changes to the API:
   library equivalents. For context, see the :ref:`Build System Breaking Changes
   <version-4.0-build-system-breaking>` section.
 
+- Removes the following deprecated ``utf8`` types and methods that have ``string`` equivalents:
+
+  - ``k_utf8`` from the ``bsoncxx::v_noabi::type`` class. Instead, use ``k_string``.
+  - ``b_utf8`` from the ``bsoncxx::v_noabi::types`` class. Instead, use ``b_string``.
+  - ``get_utf8()`` from the ``bsoncxx::v_noabi::document::element``, ``bsoncxx::v_noabi::array::element``,
+    and ``bsoncxx::v_noabi::types::bson_value::view`` classes. Instead, use ``get_string()``.
```
Comment (eramongodb) [inline source/upgrade.txt:159]: This is also an ABI breaking change. Suggest adding the following entry below:

```
- Removes deprecated ``utf8`` functions that have ``string`` equivalents. For context, see the <reference to ABI breaking changes> section.
```

File: source/upgrade.txt [hunk 4]
```diff
@@ -151,6 +151,17 @@ The v4.0 driver introduces the following breaking changes to the API:
   library equivalents. For context, see the :ref:`Build System Breaking Changes
   <version-4.0-build-system-breaking>` section.
 
+- Removes the following deprecated ``utf8`` types and methods that have ``string`` equivalents:
```
Comment (eramongodb) [inline source/upgrade.txt:154]:
```suggestion
- Removes the following deprecated ``utf8`` types and functions that have ``string`` equivalents:
```
"Method" is not a well-defined term in C++.

**Review by eramongodb — CHANGES_REQUESTED** [review]

File: source/upgrade.txt [hunk 5]
```diff
@@ -161,6 +174,10 @@ The v4.0 driver introduces the following breaking changes to the ABI:
   library equivalents. For context, see the :ref:`Build System Breaking Changes
   <version-4.0-build-system-breaking>` section.
 
+- Removes deprecated ``utf8`` types and functions that have ``string`` equivalents.
```
Comment (eramongodb) [inline source/upgrade.txt:177]:
```suggestion
- Removes deprecated ``utf8`` functions that have ``string`` equivalents.
```
Types are not part of the ABI. Only variables and functions are part of the ABI.

**Review by eramongodb — APPROVED** [review]
One feedback remaining; otherwise, LGTM.

**Pull request merged**